src/System.Linq/src/System/Linq/Concat.cs

```diff
@@ -194,10 +194,11 @@ internal override ConcatIterator<TSource> Concat(IEnumerable<TSource> next)
             internal void CopyTo(TSource[] array, int arrayIndex)
             {
                 Debug.Assert(array != null);
+                Debug.Assert(arrayIndex >= 0);
                 Debug.Assert(array.Length - arrayIndex >= Count);
 
-                _first.CopyTo(array, 0);
-                _second.CopyTo(array, _first.Count);
+                _first.CopyTo(array, arrayIndex);
+                _second.CopyTo(array, arrayIndex + _first.Count);
```

Comment:
@stephentoub Currently, this method is only called from one place: [here](https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/Concatenate.cs#L342). `indexAfterCopy - copied` will always be 0 since (as verified by the preceding assert) `copied == Count`. `CopyBefore` in turn is only called by [here](https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/Concatenate.cs#L382), which passses in `result.Length` which is equivalent to Count.

If you were looking for a less technical explanation, it's because we only copy the first 2 collections to the beginning of the array for `ToArray`.